### ISSUE TITLE
Add support for HTTP3 to the Networking API

### ISSUE THREAD

**Issue by Duckwhale** [evo-lua/evo-luvi#135]
Also needs QUIC support. Probably a ton of work for API design and library integration alone...

Currently blocked by OpenSSL/lack of QUIC libraries? Will have to revisit later, probably.

Background information:

* https://daniel.haxx.se/blog/2019/01/21/quic-and-missing-apis/
* https://daniel.haxx.se/blog/2021/10/25/the-quic-api-openssl-will-not-provide/

RFCs:

* https://www.rfc-editor.org/rfc/rfc8999.html
* https://www.rfc-editor.org/rfc/rfc9000.html
* https://www.rfc-editor.org/rfc/rfc9001.html
* https://www.rfc-editor.org/rfc/rfc9002.html

QUIC libraries:

* https://github.com/microsoft/msquic
* https://github.com/google/quiche
* https://github.com/cloudflare/quiche

It's clear that this would be a rather extensive task, and considering it's not immediately required in my own apps right now I'll postpone it until such time that it gets better support by third parties. I haven't even looked at libuv's UDP API much, so  it's hard to estimate.
